pilot/pkg/config/kube/ingressv1/conversion.go

```diff
@@ -240,6 +242,22 @@ func ConvertIngressVirtualService(ingress knetworking.Ingress, domainSuffix stri
 	}
 }
 
+// getMatchURILength returns the length of matching path, and whether the match type is EXACT
+func getMatchURILength(match *networking.HTTPMatchRequest) (length int, exact bool) {
+	uri := match.GetUri()
+	switch uri.GetMatchType().(type) {
+	case *networking.StringMatch_Exact:
+		return strings.Count(uri.GetExact(), "/"), true
+	case *networking.StringMatch_Prefix:
+		return strings.Count(uri.GetPrefix(), "/"), false
```

Comment:
Is it the number of `/` or the raw `len(uri.GetPrefix())` we want? Can we look at some others like nginx and countor and see what they do?

Reply:
Oh yes, I've checked [ingress-nginx doc](https://kubernetes.github.io/ingress-nginx/user-guide/ingress-path-matching/#example_1), should use the raw length.
Fixed